### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by jhamman**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

closes https://github.com/conda-forge/zarr-feedstock/issues/76

### CODE REVIEW

**Comment by jhamman** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -12,20 +12,20 @@ source:
...
 build:
-  number: 0
+  number: 1
   script: {{ PYTHON }} -m pip install . --no-deps -vv
   noarch: python
 
 requirements:
   host:
-    - python >=3.5
+    - python >=3.8
```
Comment (jakirkham) [inline recipe/meta.yaml:21]: Just highlighting for review. The change here seems reasonable

File: recipe/meta.yaml [hunk 2]
```diff
@@ -12,20 +12,20 @@ source:
...
     - pip
-    - setuptools >=38.6.0
+    - setuptools >=64.0.0
     - setuptools_scm >1.5.4
   run:
     - python >=3.5
```
Comment (jakirkham) [inline recipe/meta.yaml:26]: Think we want this one lined up

```suggestion
    - python >=3.8
```

**Review by jakirkham — COMMENTED** [review]
Thanks Joe! 🙏

Had a comment below

Also would you like to add yourself to the maintainers here?

File: recipe/meta.yaml [hunk 3]
```diff
@@ -12,20 +12,20 @@ source:
...
   noarch: python
 
 requirements:
   host:
-    - python >=3.5
+    - python >=3.8
     - pip
-    - setuptools >=38.6.0
+    - setuptools >=64.0.0
     - setuptools_scm >1.5.4
```
Comment (jakirkham) [inline recipe/meta.yaml:24]: Noticed we are missing this upstream (it was dropped accidentally when moving to `pyproject.toml`. Fixing with PR ( https://github.com/zarr-developers/zarr-python/pull/1477 ). Nothing to do here

**Comment by jakirkham** [comment]
@conda-forge-admin, please re-render

File: recipe/meta.yaml [hunk 4]
```diff
@@ -12,29 +12,29 @@ source:
...
 
 test:
   requires:
     - pytest
-    # Needs `pkg_resources` in the tests
-    # xref: https://github.com/zarr-developers/zarr-python/pull/813
-    - setuptools
+    # Needs `packaging` in the tests
+    # xref: https://github.com/zarr-developers/zarr-python/pull/1450
+    - packaging
```
Comment (jakirkham) [inline recipe/meta.yaml:37]: Noticed this changed in PR ( https://github.com/zarr-developers/zarr-python/pull/1450 ). So refreshed the dependency (and comment)

Comment (jakirkham) [reply recipe/meta.yaml:37]: Ah it's not in 2.15 🤦‍♂️

Will revert and add to the release PR

**Comment by jakirkham** [comment]
Thanks Joe! 🙏 

Sorry for the churn at the end. Should be good to go once CI passes 🙂

File: recipe/meta.yaml (hunk 4 above)
Comment (jakirkham) [reply recipe/meta.yaml]: Adding here ( https://github.com/conda-forge/zarr-feedstock/pull/80#discussion_r1270089929 )

**Pull request merged**